test/expect-tests/dune_console/dune_console_tests.ml

```diff
@@ -0,0 +1,104 @@
+         time that this is over."
+    ]
+;;
+
+let test_status_line_clearing (module Console : New_console) =
+  let open Console in
+  Status_line.set (Status_line.Constant (Pp.text "Here is a status line"));
+  Status_line.set (Status_line.Constant (Pp.text "Here is another"));
+  Status_line.clear ()
+;;
+
+let test_status_line_clearing_with_wrapping (module Console : New_console) =
+  let open Console in
+  Status_line.set
+    (Status_line.Constant
+       (Pp.text
+          "This status line is a problem because of the fact that it is especially long \
+           and therefore will not be cleared properly."));
+  Status_line.set (Status_line.Constant (Pp.text "Here is another"));
+  Status_line.clear ()
+;;
+
+(** Dumb backend *)
+
+let%expect_test "basic usage" =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.dumb;
+  test_basic_usage (module Console);
+  [%expect
+    {|
+    Hello World!
+    Hello this is a very long sentence that will probably wrap the console by the
+    time that this is over. |}]
+;;
+
+let%expect_test "Status line clearing." =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.dumb;
+  test_status_line_clearing (module Console);
+  [%expect "\nHere is a status line\nHere is another"]
+;;
+
+let%expect_test "Status line clearing with wrapping." =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.dumb;
+  test_status_line_clearing_with_wrapping (module Console);
+  [%expect
+    "\n\
+     This status line is a problem because of the fact that it is especially long\n\
+     and therefore will not be cleared properly.\n\
+     Here is another "]
+;;
+
+(** Progress backend *)
+
+let%expect_test "basic usage" =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.progress;
+  test_basic_usage (module Console);
+  [%expect
+    {|
+    Hello World!
+    Hello this is a very long sentence that will probably wrap the console by the
+    time that this is over. |}]
+;;
+
+let%expect_test "Status line clearing." =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.progress;
+  test_status_line_clearing (module Console);
+  [%expect
+    "Here is a status line\r                     \rHere is another\r               \r"]
+;;
+
+(* CR-someday alizter: this should insert the appropriate number of "\r"s in order to
+   fully clear the previous lines when wrapped. *)
+let%expect_test "Status line clearing with wrapping." =
+  let module Console = New () in
+  Console.Backend.set Console.Backend.progress;
+  test_status_line_clearing_with_wrapping (module Console);
+  [%expect
+    "\n\
+     This status line is a problem because of the fact that it is especially long and \
+     therefore will not be cleared \
+     properly.\r                                                                                                                        \
```

Comment:
cc @Julow You may be interested to know that the formatting here is a bit whack. I would expect that a new `\` is inserted after `\r` and that the white space is also wrapped.

Reply:
Would you expect something like this ?

```
  [%expect
    "\n\
     This status line is a problem because of the fact that it is especially long and \
     therefore will not be cleared \
     properly.\r                                                                        \
                                                     \n\
     Here is another\r               \r "]
```

Reply:
Yes, that seems more reasonable. Just wanted to give you a heads up, this test won't be around for long.

Reply:
Ok, thanks!